portal/templates/portal_wrapper.html

```diff
@@ -94,7 +94,7 @@
                          <button id="tnthUserBtn" type="button" class="tnth-nav-btn tnth-white tnth-dropdown-toggle" data-toggle="tnth-dropdown">
                             {% if enable_links %}&nbsp;&nbsp; {{ _("MENU") }} <i class="fa fa-bars"></i>{% endif %}
                             <img class="profile-img" src="{{movember_profile}}" width="50" height="50" alt="{{_('Profile image')}}"/>
-                            <span class="welcome-text"><em>{{ _("Welcome") }}</em>{% if user.display_name != "Anonymous" %}, {{ user.display_name }}{% endif %}</span>
+                            <span class="welcome-text portal-weak-auth-hide"><em>{{ _("Welcome") }}</em>{% if user.display_name != "Anonymous" %}, {{ user.display_name }}{% endif %}</span>
```

Comment:
Would a check for `user.has_role('write_only')` be more appropriate than checking a symptom of that role?

Reply:
think the issue here that we don't check the user's role, but checking the authentication method, i.e. url_authenticated, in the user's encounter instead

Reply:
@achen2401 is correct.  roles are generally static, as opposed to an encounter level auth method.

Reply:
Ah, will the check in the view also work in the template?:
`user.current_encounter.auth_method == 'url_authenticated'`

Reply:
thanks, ivan, I think it works, although, in this case we want to hide the relevant display `user.current_encounter.auth_method != 'url_authenticated'`